tracardi/process_engine/action/v1/notification_plugin_action.py

```diff
@@ -0,0 +1,152 @@
+
+from tracardi.domain.resource import ResourceCredentials
+from tracardi.service.plugin.domain.config import PluginConfig
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.runner import ActionRunner
+
+
+class Config(PluginConfig):
+    source: NamedEntity
+    template_name: str
+    subscriber_id: str
+    recipient_email: Optional[str] = ""
+    payload: Optional[str] = "{}"
+
+
+def validate(config: dict) -> Config:
+    return Config(**config)
+
+
+class NotificationGeneratorAction(ActionRunner):
+
+    @staticmethod
+    async def build(**kwargs) -> 'NotificationGeneratorAction':
+        config = validate(kwargs)
+        resource = await storage.driver.resource.load(config.source.id)
+        return NotificationGeneratorAction(config, resource.credentials)
+
+    def __init__(self, config: Config, credentials: ResourceCredentials):
+        self.config = config
+        self.credentials = credentials.get_credentials(self, output=Token)  # type: Token
+
+    async def run(self, payload: dict, in_edge=None) -> Result:
+        url = 'https://api.novu.co/v1/events/trigger'
+        dot = self._get_dot_accessor(payload)
+        timeout = aiohttp.ClientTimeout(total=15)
+        params = {
+            "name": self.config.template_name,
+            "to": {
+                "subscriberId": dot[self.config.subscriber_id],
+                "email": dot[self.config.recipient_email]
+            },
+            "payload": json.loads(self.config.payload)
+        }
+        print(params)
+        async with HttpClient(timeout=timeout, retries=self.node.on_connection_error_repeat) as client:
+            async with client.request(
+                    method='post',
+                    url=url,
+                    headers={"Authorization": f"ApiKey {self.credentials.token}",
+                             "Content Type": "application/json"},
+                    ssl=False,
+                    json=params
+            ) as response:
+                try:
+                    content = await response.json()
+                except JSONDecodeError:
+                    content = await response.text()
+
+                result = {
+                    "status": response.status,
+                    "content": content
+                }
+
+                if response.status in [200, 201, 202, 203]:
+                    return Result(port="response", value=result)
+                else:
+                    return Result(port="error", value=result)
+
+
+def register() -> Plugin:
+    return Plugin(
+        start=False,
+        spec=Spec(
+            module=__name__,
+            className='NotificationGeneratorAction',
+            inputs=['payload'],
+            outputs=['response', 'error'],
+            version="0.7.2",
+            license="MIT",
+            author="Mateusz Zitaruk",
+            init={
+                "source": {"id": "", "name": ""},
```

Comment:
tough case here - I'd put a validator on Config class, because that enables source.id to be an empty string (I think, gotta check it on frontend)